templates/openwisp2/settings.py

```diff
@@ -49,6 +49,9 @@
     'flat_json_widget',
 {% if openwisp2_network_topology %}
     'openwisp_network_topology',
+{% endif %}
+{% if openwisp2_firmware_upgrader %}
+    'openwisp_firmware_upgrader',
 {% endif %}
```

Comment:
I think this is missing:

```
PRIVATE_STORAGE_ROOT = os.path.join(BASE_DIR, 'private')
```

Where are files stored without this?

Reply:
I have used following because it has been suggested in the documentation of openwisp-firmware-upgrader. 

```
PRIVATE_STORAGE_ROOT = '{0}/firmware/'.format(BASE_DIR)
```